File: 3959-check-good-integer/3959-check-good-integer.scala
object Solution {
    def checkGoodInteger(n: Int): Boolean = {
        var num = n
        var ds = 0
        var ss = 0

        while (num > 0) {
            val curr = num % 10
            ds += curr
            ss += curr * curr
            num /= 10
        }

        ss - ds >= 50
    }
}

File: 3959-check-good-integer/3959-check-good-integer.py
class Solution(object):
    def checkGoodInteger(self, n):
        """
        :type n: int
        :rtype: bool
        """
        ds = 0
        ss = 0

        while n > 0:
            curr = n % 10
            ds += curr
            ss += curr * curr
            n //= 10

        return ss - ds >= 50